tests/integration/go.mod

```diff
@@ -0,0 +1,12 @@
+module github.com/k8ssandra/k8ssandra/tests/integration
```

Comment:
The current approach was to use a single go.mod for everything, or at least everything for every test. We don't want the unit tests and integration tests to diverge in their modules.

Reply:
That sounds a bit contrary to what we do for the separation of the tests themselves.  @burmanm can you explain a bit why if we have dependencies only for unit tests, and some for integration tests, why would we want to lump them all together making it more difficult to track specific dependencies?

Reply:
The norm is a single module per project. See https://github.com/golang/go/wiki/Modules#modules.

Reply:
With respect to test separation in k8s projects, the following structure is pretty common:

* Unit tests live in the same directory as the code under test
* Integration tests that use [envtest](https://github.com/kubernetes-sigs/controller-runtime/tree/master/pkg/envtest) also live in the same directory as the code under test; however, I think some separation might be worth considering to make it clear that these are not unit tests.
* End-to-end tests that run against a live cluster live under `<project-root>/test/e2e`.

Reply:
We can rename our mentions of `integration` to `e2e` if that complies better with Go/k8s terminology, no worries.
Anyone against doing so?

Reply:
I like the idea of keeping `unit`, `integration`, and `e2e` separate.  Something like: 

`<project-root>/test/unit` (single scoped verifications)
`<project-root>/test/integration` (not using live cluster)
`<project-root>/test/e2e` (using live cluster - more blackbox style)

Reply:
I'll remove the submodule, I thought it was something we wanted here because of dependency conflicts (although VSCode gave me a hard time with it).
I struggle a bit with the distinction between integration tests and e2e tests 🤔 
How would the tests implemented in this PR qualify to you between the two?

Reply:
My experience has been that `integration` testing is verifying an integration between two components typically.  That could be done without a live cluster and may include more mocks/spys as needed.
 
1) Testing `A+B` and `B+C` gives some confidence for `A->C` but the tests are written a bit differently in that they are testing at a lower level of integration.

2) E2E is more of a user experience something like input @ `A -> C` (verify output at end) which can be more black box in nature.

This [doc ](https://dev.to/betoyanes/unit-vs-integration-vs-e2e-tests-4b2o) is a good reference for differences at these levels.